active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,571 @@
+
+<!-- Explain why TT is not recommended for notifications --> 
+
+
+## Examples
+
+The teaching tip has several states, including these notable ones.
+
+A teaching tip can point to a specific UI element to enhance contextual clarity of the information it is presenting. 
+
+<a href="url"><img src="TeachingTipPointerSample.jpg" height="600"></a>
+
+A teaching tip can be used without a tail when the information that it is presenting is not specific to a UI element. 
+
+<a href="url"><img src="TeachingTipSampleApp.jpg" height="600"></a>
+
+A teaching tip can require manual acknowledgement before dismissing or a teaching tip may also be light-dismiss enabled so that it will dismiss when a user scrolls or interacts with other elements of the application.
+
+<a href="url"><img src="TeachingTipLightDismissSample.jpg" height="600"></a>
+
+
+## Create a teaching tip
+
+<!-- Use the Password property to get or set the contents of the PasswordBox. You can do this in the handler for the PasswordChanged event to perform validation while the user enters the password. Or, you can use another event, like a button Click, to perform validation after the user completes the text entry. -->
+
+Here's the XAML for a teaching tip control that demonstrates the default look of the TeachingTip with a title and subtitle. 
+
+When the user clicks the button, a teaching tip will appear and display a message to the user. 
+
+XAML
+```XAML
+<Button Content="Save" Click="OnFirstSaveButtonClick"></Button>
+<muxc:TeachingTip> 
+    <muxc:TeachingTip x:Name="AutoSaveTip"
+        Title="Saving automatically"
+        Subtitle="We save your changes as you go - so you never have to.">
+    </muxc:TeachingTip>
+</muxc:TeachingTip.Attach>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+Here's the result when this code runs and the clicks the button.
+
+<a href="url"><img src="TeachingTipSampleApp.jpg" height="600"></a>
+
+### Add a margin 
+
+A teaching tip can have a margin added by setting the Margin property. When a teaching tip is attached to a UI element, a margin will move it away from the UI element it is pointing to. When a teaching tip is not attached to anything, a margin will move it away from the side of the app window it is positioned towards. 
+
+XAML
+```XAML
+<Button Content="Save" Click="OnFirstSaveButtonClick"></Button>
+<muxc:TeachingTip> 
+    <muxc:TeachingTip x:Name="AutoSaveTip"
+        Title="Saving automatically"
+        Subtitle="We save your changes as you go - so you never have to."
+        Margin="80">
+    </muxc:TeachingTip>
+</muxc:TeachingTip.Attach>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipMarginSample.jpg" height="600"></a>
+
+### Point to a UI element
```

Comment:
I understand that you want to show from default to more advanced features, but I wonder if we should show this one first since isn't this the most recommended way of using this control?

Reply:
@chigy This reason is why I showed pointing teaching tips as the first example under the Examples section. 

In the recommended guidance samples, such as Password Box, everything under the "Create a..." heading appears to go 1) simplest to most fully featured, followed by 2) other options that are _somewhat_ regular, and last, 3) advance scenario examples. 

But this pattern is simply what I inferred. @mijacobs, do you have a recommended approach?

Reply:
@SavoySchuler @mijacobs - This is a good point that we should have a guidance for guidance to spec writers so we are consistent.

Reply:
It is also worth noting that **Examples** calls out the suggested approach of using pointing tips for teaching moments that refer to specific UI elements and non-pointing tips for scenarios that do not refer to specific UI elements. **Recommendations** should call this out too. I will add it.

Reply:
I think what's confusing is that it starts with the most basic case, but that example doesn't have code.  The first *code* example is not the most basic.  Why not have code for the first ones?

Reply:
@MikeHillberg Could you point me to the example you are looking at to help me make sure I am on the same page? 

Looking at Password Box as a reference, its Guidelines open with three codeless functional examples of the control to show the reader some broad points of key functionality. I imagine this helps answer the survey level question of "Is this the right control?" before moving into the examples with code. It is worth noting, the three opening examples are also shown later with code.

Reply:
Pinging @chigy, @MikeHillberg, and @mijacobs on this question of **Examples** ordering as I am working to close the guidelines and documentation section soon. I followed the formatting set by PasswordBox's Examples section that shows the recommend usage first but then moves from simple to advanced code samples. The comments here make me believe there may be a more preferred way to order this section, however, I am unclear on the actionable solution that is being suggested.

Reply:
This is just my opinion and suspect @MikeHillberg and @mijacobs might have different opinions...
1) use the most basic least code control use - assumption here is that if you speced your control in a meaningful way, this should present the developer with the highest use case
2) common usage scenarios that require some code writing
3) custom solutions that may or may not be needed by all but still worth mentioning

Reply:
Thanks, @chigy! I believe that is the order I have it in, but please let me know if you see it differently? I pointing tips are the brand image for the control because it best distinguishes what the control is without context, but believe non-pointing tips will be equally common and are more simple to implement, which is why I ordered their examples first. I am happy to reorder the examples with pointing tips first (this would align branding exactly) and market the non-pointing tips as "removing" the pointer rather than "adding" it.

I am only concerned that it may deter customers who are looking for the simpler, non-pointing tips as it is atypical of controls to get simpler throughout the examples as opposed to more complex. Thoughts? 

Surfacing again for @MikeHillberg and @mijacobs.

Reply:
The reason that I find the untargeted example to be more complicated than the targeted example -- which I agree seems counterintuitive -- is that I thought the untargeted scenario would be less common.  So it's not that we show the simple example first, we show the typical example first.  But if they're 50/50 scenarios, then agree with simpler first.

Reply:
@SavoySchuler , IMO the best representation of TeachingTip is with the pointing tip (we even have "tip" in the name? :) ) and that's what I see many apps using it today. So I feel it should be the first example? That said, you are the PM owner of this control who did most of the investigation, so if you tell me that's not right, then I'm happy either way.

Reply:
I think I am over indexing on non-pointing tips due what I have seen with the sample of our early adopters. You are right that when we look at the broader spectrum of application (or even just Microsoft Office), pointing tips out number non-pointing tips as a fact. Not completely - but certainly enough that it is inaccurate to label the spread 50/50. I will make this change. Thanks, everyone!